kornia/enhance/adjust.py

```diff
@@ -44,7 +45,43 @@ def adjust_saturation_raw(input: torch.Tensor, saturation_factor: Union[float, t
     return out
 
 
-def adjust_saturation(input: torch.Tensor, saturation_factor: Union[float, torch.Tensor]) -> torch.Tensor:
+def adjust_saturation_pillow_raw(input: torch.Tensor, saturation_factor: Union[float, torch.Tensor]) -> torch.Tensor:
+    r"""Adjust color saturation of an image as pillow."""
+
+    if not isinstance(input, torch.Tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(saturation_factor, (float, torch.Tensor)):
+        raise TypeError(
+            f"The saturation_factor should be a float number or torch.Tensor." f"Got {type(saturation_factor)}"
+        )
+
+    if isinstance(saturation_factor, float):
+        saturation_factor = torch.as_tensor(saturation_factor)
+
+    saturation_factor = saturation_factor.to(input.device).to(input.dtype)
+
+    # TODO: find a proper way to check bound values in batched tensors.
+    # if (saturation_factor < 0).any():
+    #     raise ValueError(f"Saturation factor must be non-negative. Got {saturation_factor}")
+
+    for _ in range(len(input.shape) - len(saturation_factor.shape)):
+        saturation_factor = torch.unsqueeze(saturation_factor, dim=-1)
+
+    x_other: torch.Tensor = rgb_to_grayscale(input, torch.tensor([0.2989, 0.587, 0.114]))
```

Comment:
we need a different logic here -- kornia it's not only rgb images
https://github.com/kornia/kornia/pull/1586/files#diff-ff1ae62fe6ba5559b9ad03f2ad9006d1c42c4b035fb7016a861c27251a523df8R354

Reply:
In the spirit of having a mode that specifies we are following pillow standard, what do you think about checking that the img has one channel or three and raising an exception otherwise like in [torchvision](https://github.com/pytorch/vision/blob/d8654bb0d84fd2ba8b42cd58d881523821a6214c/torchvision/transforms/functional_tensor.py#L179) ?

Reply:
i don't think it's a good idea to raise an exception here. In fact, there reason why we do different is because we couldn't find any solid reference why pillow implements in that way. Then we looked at the book definition of Szeliski. I'd be interested in seeing what other libraries do other than torchvision -- the plan is to offer others backends than in the midterm relatively soon.

Reply:
What I thought about is to raise the exception in the `adjust_saturation_pillow_raw` function. I think it would be the expected behavior as this function is only called if the user explicitly asked for the pillow backend that raises an exception if the input is not an RGB image.

Reply:
For contrast, there is the same issue I also followed torchvision/pillow implementation that uses rgb so I think that I should also raise an exception if the image is not RGB or 1-channel if the user explicitly asks for pillow backend.

Reply:
ok but i'd avoid adding the keyword to another library to the high level call. Maybe just having the low level  `adjust_saturation_additive_rgb` with explicit docs where you strictly replicate whatever pillow implements  /cc @shijianjian

Reply:
I jusr pushed a new commit that separate to modes brightness, contrast and saturation in order to remove `pillow_like` flag.

Reply:
hi ! thanks -- we were about to merge https://github.com/kornia/kornia/pull/1586 where I included a
`adjust_constrast_with_mean_substraction` that replicates what pil does as agreed with @ducha-aiki 

lets merge somehow both PR so that from the high level augmentations api can be reproducible pytorch codes.
Check the comments on the docs we added there. Our goal as a library is to somehow be compatible with pytorch but not strictly follow their conventions. Specially, since they also support tensor augmentations.

https://github.com/kornia/kornia/pull/1586/commits/fc4c48546a2dd23378171baa893ec1cdc795d020#diff-ff1ae62fe6ba5559b9ad03f2ad9006d1c42c4b035fb7016a861c27251a523df8R311

Reply:
Thanks for your feedback. Should I wait your merge, then update this pull request to follow your lead and update this pull request accordingly ot what you did?

Reply:
I think yes -- i also added few utilities for testing that are quite useful.